src/Helpers/LeanCode.UserIdExtractors/IUserIdExtractor.cs

```diff
@@ -0,0 +1,14 @@
+using System.Security.Claims;
+
+namespace LeanCode.UserIdExtractors;
+
+public interface IUserIdExtractor<TUserId> : IUserIdExtractor
+    where TUserId : notnull, IEquatable<TUserId>
+{
+    TUserId ExtractId(ClaimsPrincipal user);
+}
+
+public interface IUserIdExtractor
+{
+    string Extract(ClaimsPrincipal user);
```

Comment:
Yea it's hiding base method otherwise, we can either keep both methods with different names or do not inherit `IUserIdExtractor` in `IUserIdExtractor<TUserId>` at all.